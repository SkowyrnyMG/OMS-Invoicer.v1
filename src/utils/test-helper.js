import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
