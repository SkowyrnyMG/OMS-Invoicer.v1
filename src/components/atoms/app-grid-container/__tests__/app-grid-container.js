import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import AppGridContainer from '../app-grid-container';

describe('AppGridContainer', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<AppGridContainer />);
  });
});
