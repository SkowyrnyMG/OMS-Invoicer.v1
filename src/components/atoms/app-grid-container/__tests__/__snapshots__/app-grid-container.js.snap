// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppGridContainer should display in the document with correct styles 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 65vw min-content;
  grid-template-rows: min-content;
  grid-column-gap: 2rem;
}

@media (max-width:1300px) {
  .c0 {
    grid-template-columns: 63vw min-content;
  }
}

@media (max-width:1168px) {
  .c0 {
    grid-template-columns: 73vw min-content;
  }
}

@media (max-width:767px) {
  .c0 {
    grid-template-columns: 90vw;
  }
}

<div
  class="c0"
/>
`;
